llarp/dns/server.cpp

```diff
@@ -120,12 +127,11 @@ namespace llarp::dns
 
     if (m_QueryHandler and m_QueryHandler->ShouldHookDNSMessage(msg))
       return true;
-
-    if (m_Resolvers.find(to) != m_Resolvers.end())
-    {
-      return false;
-    }
-    return true;
+#ifdef ANDROID
+    return IsUpstreamResolver(to, from);
+#else
```

Comment:
This looks suspicious to me:

When android (configured with random DNS server) makes a DNS request, we want to intercept it, hand it over to unbound, then unbound makes the request to the lokinet-configured upstream.

So we want this to return true on the first request (to intercept) and false on the second request (to let the upstream request through).

And that is what the `#else` case is going to do; the logical inversion for android doesn't seem right.

(Either way, this needs a comment.  I suggest:
```C++
// If this request is going to an upstream resolver then we want to let it through (i.e. don't handle it), and so
// want to return false.  If we have something else then we want to intercept it to route it through our caching
// libunbound server (which then redirects the request to the lokinet-configured upstream, if not cached).
```